### PR TITLE
Add support for Unix sockets

### LINKED ISSUE #160: support UNIX sockets for server and viewer
Hi.

It would be nice if the connection would work via UNIX sockets, for both the server and the client.

This would be a great security boost, since on a local system, connection to the server socket can be controlled via normal file permissions (which isn't possible for inet sockets), and to the remote the socket can be forwarded via SSH.

Cheers,
Chris.
## 

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/10254497-support-unix-sockets-for-server-and-viewer?utm_campaign=plugin&utm_content=tracker%2F3557444&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F3557444&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by CendioOssman**
Fixes #160 for the FLTK viewer and the Unix servers.

### CODE REVIEW

File: unix/xserver/hw/vnc/vncExtInit.cc [hunk 1]
```diff
@@ -79,6 +80,8 @@ rfb::IntParameter httpPort("httpPort", "TCP port to listen for HTTP",0);
 rfb::AliasParameter rfbwait("rfbwait", "Alias for ClientWaitTimeMillis",
                             &rfb::Server::clientWaitTimeMillis);
 rfb::IntParameter rfbport("rfbport", "TCP port to listen for RFB protocol",0);
+rfb::StringParameter rfbpath("rfbpath", "Unix socket to listen for RFB protocol", "");
+rfb::IntParameter rfbmode("rfbmode", "Unix socket access mode", 0600);
```
[the remark this answers is not in the document]
Comment (CendioOssman) [reply unix/xserver/hw/vnc/vncExtInit.cc]: `rfbunixpath`/`rfbunixmode`?

[the remark this answers is not in the document]
Comment (astrand) [reply unix/xserver/hw/vnc/vncExtInit.cc]: +1

**Comment by CendioOssman** [comment]
The comments should hopefully be addressed. Please have a new look.

File: common/network/UnixSocket.cxx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+  if (getsockname(getFd(), (struct sockaddr *)&addr, &salen) != 0) {
+    vlog.error("unable to get local name for socket");
+    return rfb::strDup("");
+  }
+
+  if (salen > offsetof(struct sockaddr_un, sun_path))
+    return rfb::strDup(addr.sun_path);
+
+  vlog.error("unable to get name for socket");
+  return rfb::strDup("");
```
Comment (astrand) [inline common/network/UnixSocket.cxx]: Last two lines needs a tweak to better handle unnamed sockets created from socketpair(): Skip the error message, and return something better than the empty string. (x11clone currently uses "UNIX Socket")

**Pull request merged**